chapter.md

```diff
@@ -0,0 +1,781 @@
+That will make it easier to swap things in and out for testing later on.
+(It also encourages us to avoid using `this`,
+which in turn encourages us to think more functionally.)
+
+```ts
+export interface IForm {
+  character: ICharacter;        // our single top-level object (for now)
+}
+
+export interface ICharacter {
+  name?: string;                // optional name
+  bioSummary: IBioSummary;      // biographical information (see below)
+  skills: string[];             // list of skills
+}
+
+export interface IBioSummary {
+  age: number;                  // age in years
+  alignment: string;            // good or bad, lawful or chaotic
+  race: string;                 // character's species
+}
+```
+
+**FIXME: @renvrant https://github.com/renvrant/formcontrol-freaks/pull/3#discussion_r106822417**
+
+These interfaces define a store
+that contains a single object representing a character,
+rather than using the character itself as the store.
+That way,
+if we want to add other top-level items in future,
+we won't need to reorganize existing material.
+For example,
+the next version of our application might have this structure:
+
+```ts
+export interface IForm {
+  character: ICharacter;        // character info
+  equipment: IEquipment;        // character's equipment
+}
+```
+
+Equally,
+if we want to add more forms,
+we won't need to reorganize the existing material,
+and we can re-use common actions that work for any form.
+
+One consequence of this decision is that
+every action should contain the path to the particular part of the state it applies to.
+Agile purists might say that we shouldn't add this until we need it,
+and if our application consisted of just a login form and a message of the day,
+they'd be right.
+However,
+we have enough experience with applications of this kind to know that we're going to,
+and adding that to our architecture from the start will save us re-work later.
+
+Having decided on the structure of our store,
+the next step is to define its initial state.
+We will do this by creating a literal object that conforms to the `ICharacter` interface,
+and another that conforms to the `IForm` interface:
+
+```ts
+const characterInitialState: ICharacter = {
+  name: undefined,
+  bioSummary: {
+    age: undefined,
+    size: undefined,
+    alignment: undefined,
+    race: undefined,
+  },
+  skills: []
+};
+
+const initialState: IForm = {
+  character: characterInitialState
+};
+```
+
+We could fold the definition of `characterInitialState` into `initialState`
+to create one large literal object,
+but again,
+since we're likely to add more state information
+(like the equipment the character is carrying),
+we have decided that `initialState` will only ever be
+a list of top-level objects conforming to separately-defined interfaces.
+
+> We have defined our initial state in a file of its own
+> to make it easier to find and update.
+> In a real application,
+> it would probably be created by some sort of factory
```

Comment:
I switched "by some sort of factory" for "programmatically" (which is about as anodyne as it can be).  I think the extra detail will be a distraction here, since we're already in a callout block. Thoughts?